apps/ima/fields/index.js

```diff
@@ -6,5 +6,166 @@ module.exports = {
     mixin: 'radio-group',
     options: ['person-named', 'has-legal-representative', 'someone-else'],
     validate: 'required'
+  },
+  'how-removal-condition-1-applies': {
+    mixin: 'checkbox-group',
+    validate: 'required',
+    options: [{
+      value: 'entered-without-permission'
+    }, {
+      value: 'by-deception'
+    }, {
+      value: 'deportation-order-applies'
+    }, {
+      value: 'travel-ban'
+    }, {
+      value: 'arrived-without-visa'
+    }, {
+      value: 'arrived-without-eta'
+    }]
+  },
+  'entered-without-permission': {
+    isPageHeading: false,
+    mixin: 'radio-group',
+    options: [{
+      value: 'yes'
+    }, {
+      value: 'no',
+      toggle: 'entered-without-permission-detail-fieldset',
+      child: 'partials/entered-without-permission-detail'
+    }],
+    validate: 'required'
+  },
+  'entered-without-permission-detail': {
+    mixin: 'textarea',
+    validate: ['required', 'notUrl', { type: 'regex', arguments: /^[^\[\]\|<>]{1,15000}$/ }],
```

Comment:
what is the regex doing?

Reply:
On top of the 15,000 character limit, it prevents the [ ] | < > characters from being entered because they caused issues in another project, so I included the check in here too. As far as I remember, it was due to nginx throwing an error for those characters.

Reply:
Good idea man.  The thing is, they are different errors.  
One is to do with maxlength so you can be specific.  In HOF, we have a maxlength validation, you should use that as that is the standard.  Also it comes with a GDS design which shows the amount of characters left.
https://design-system.service.gov.uk/components/character-count/

The other part is about the specific characters entered and you can put that as regex, with a message specific about the characters, "The characters you enters must not include '<', '>', '[', ']'".  It's probably worth asking the content guys what is best for that".

Also, do you want to maybe add this as part of our validation list in HOF so it can reusable for others to use as a separate ticket?

Reply:
From the ETA project, it was highlighted that the 'maxlength' property was not accessible, so I created that solution as a workaround, which meant I had to combine the two validation messages (max length and special characters) into one. If we had to separate the messages and also keep it accessible, it might need a new validation property to be added in the hof framework. I'll confirm with the content designers if combining the two validation messages into one is okay. 

Also, it's mentioned in [the story](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-53) description that the character limit should not be shown on the field, and is missing in the Figma design. Probably another point that needs confirming with the designers.

Reply:
Ahh ok, why isn't the maxlength accessible? We were using a slightly older gov uk design system so 

The validators are here https://github.com/UKHomeOfficeForms/hof/blob/dee45923ee811c97885b2c396ae80be12c2cfd9d/controller/validation/validators.js#L17

Either way, I wouldn't be putting two errors together: length is different to the type of characters.  It will be confusing to the users.  
We want to try to improve the framework rather than do workarounds like this.

Reply:
Based on our discussion, I've separated out the maxlength and regex validations to have two different validation messages.